portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,17 +38,36 @@
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  // Whitelist to avoid the possibility of XSS
-  private final String UNSAFE_CHARACTERS_REGEX = "[^A-Za-z0-9._~()'!*:@,;+?\\s-]";
-
   DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
   private final Gson gson = new Gson();
 
+  /**
+   * @return Returns the request parameter associated with the inputted name,
+   * or returns the default value if the specified parameter is not defined.
+   */
+  public static String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query allCommentsQuery = 
-        new Query("Comment").addSort(Comment.TIME_POSTED_KEY, SortDirection.DESCENDING);
-    List<Comment> comments = queryCommentsDatastore(allCommentsQuery);
+    ValidationResult<Pagination> paginationResult = Pagination.getIncomingPagination(request);
+
+    if (paginationResult.hasValidationError()) {
+      sendRawTextError(
+          response, HttpServletResponse.SC_BAD_REQUEST, paginationResult.getValidationMessage());
+      return;
+    }
+
+    Query allCommentsQuery = new Query("Comment")
```

Comment:
We could even call `allCommentsQuery` `sortedCommentsQuery` which I think is a more accurate name

Reply:
Agreed. Changing.